QuadradiusR/scripts/torus.gd

```diff
@@ -61,27 +62,26 @@ func _on_mouse_event(viewport: Node, event: InputEvent, shape_idx: int):
 
 
 func _begin_drag():
+	anim.play("Pickup")	
 	is_held = true
-	pickup_sfx.play()
-	shadow.visible = false
+	
 	get_tree().call_group("torus", "set_interaction", false)
 	get_tree().call_group("board", "_torus_pickup", self)
 	get_tree().call_group("tile", "_torus_pickup", current_tile)
-	yield(get_tree(), "idle_frame")  # needed for the scaling to work properly
-	self.rect_scale = Vector2(1.5, 1.5)
+
 	set_process(true)
 
 
 func _end_drag():
 	set_process(false)
-	shadow.visible = true
+	anim.play("Putdown")
+
 	get_tree().call_group("torus", "set_interaction", true)
 	get_tree().call_group("board", "_torus_putdown", self)
 	get_tree().call_group("tile", "_torus_putdown", current_tile)
 	
 	is_held = false
-	putdown_sfx.play()
-	self.rect_scale = Vector2(1, 1)
+
 	rect_position = self.starting_pos
```

Comment:
So now dragging a torus is an animation, not changing torus properties, right?

Reply:
Yes. The animation player handles all visual aspects so that we don't need to do it in code. As a bonus we get smooth animation instead of just changing scale